### PR TITLE
enable SDL validation

### PR DESCRIPTION
**Pull request by AdamYoblick**
/cc: @merriemcgaw @SusLes

### CODE REVIEW

**Comment by mmitche** [comment]
This should be enabled outside of the build, otherwise it will end up on the critical path for build time. See https://github.com/dotnet/winforms/pull/3551 for how it was done for winforms. This will enable validation in two places:
- Nightly validation
- Validation on each release candidate.

**Comment by AdamYoblick** [comment]
Alright, thanks. I’ll get it done :)

________________________________

File: eng/stages/publish.yml [hunk 1]
```diff
@@ -17,6 +17,20 @@ stages:
     symbolPublishingAdditionalParameters: "'-warnAsError:$false'"
     publishInstallersAndChecksums: true
 
+    SDLValidationParameters:
+        enable: true
+        params: ' -SourceToolsList @("policheck","credscan")
+        -TsaInstanceURL https://devdiv.visualstudio.com/
```
Comment (dagood) [inline eng/stages/publish.yml:23]: IMO `>-` is nicer, lets you format it without awkward indent and the trailing `'`, e.g.:

https://github.com/dotnet/core-setup/blob/d98d0f7070a6985771054b7a4623eb06dceddb47/eng/stages/publish.yml#L25-L35
```yml
      params: >-
        -SourceToolsList @("policheck","credscan")
        -TsaInstanceURL "$(TsaInstanceURL)"
        -TsaProjectName "$(TsaProjectName)"
        -TsaNotificationEmail "$(TsaNotificationEmail)"
        -TsaCodebaseAdmin "$(TsaCodebaseAdmin)"
        -TsaBugAreaPath "$(TsaBugAreaPath)"
        -TsaIterationPath "$(TsaIterationPath)"
        -TsaRepositoryName "$(TsaRepositoryName)"
        -TsaCodebaseName "$(TsaCodebaseName)"
        -TsaPublish $True
```

http://yaml-multiline.info/ is nice to explore.

Comment (AdamYoblick) [reply eng/stages/publish.yml:23]: Nice! Will fix :)

**Pull request merged**